src/libraries/System.Linq.Parallel/tests/ExchangeTests.cs

```diff
@@ -88,10 +89,15 @@ public static IEnumerable<object[]> AllMergeOptions_Multiple()
         // The basic tests are covered elsewhere, although without WithDegreeOfParallelism
         // or WithMergeOptions
 
-        [Theory]
+        [ConditionalTheory]
         [MemberData(nameof(PartitioningData), new[] { 0, 1, 2, 16, 1024 })]
         public static void Partitioning_Default(Labeled<ParallelQuery<int>> labeled, int count, int partitions)
         {
+            if (partitions > 1 && !PlatformDetection.IsThreadingSupported)
+            {
+                throw new SkipTestException(nameof(PlatformDetection.IsThreadingSupported));
```

Comment:
This pattern seems pretty common across the various PRs being opened in this realm... maybe we should add a:
```C#
public static void ThrowSkipExceptionIfThreadingNotSupported();
```
to PlatformDetection.

Reply:
Right now it's only four cases across all the PRs we did so I feel like we don't need it yet.